File: iosApp/Shared/Screens/More/ActivityHistoryScreen.swift
//
//  ActivityHistoryScreen.swift
//  Well
//
//  Created by Phil on 12.02.2022.
//  Copyright © 2022 Well. All rights reserved.
//

import SwiftUI
import SharedMobile

private typealias Feature = ActivityHistoryFeature

struct ActivityHistoryScreen: View {
    let state: ActivityHistoryFeature.State
    let listener: (ActivityHistoryFeature.Msg) -> Void

    var body: some View {
        NavigationBar(
            title: state.title,
            leftItem: .back {
                listener(Feature.MsgBack())
            }
        )
    }
}

File: modules/androidUi/src/main/kotlin/composableScreens/more/ActivityHistoryScreen.kt
package com.well.modules.androidUi.composableScreens.more

import com.well.modules.androidUi.customViews.ControlItem
import com.well.modules.androidUi.customViews.NavigationBar
import com.well.modules.features.more.moreFeature.subfeatures.ActivityHistoryFeature.Msg
import com.well.modules.features.more.moreFeature.subfeatures.ActivityHistoryFeature.State
import androidx.compose.foundation.layout.ColumnScope
import androidx.compose.runtime.Composable

@Composable
fun ColumnScope.ActivityHistoryScreen(
    state: State,
    listener: (Msg) -> Unit,
) {
    NavigationBar(
        title = state.title,
        leftItem = ControlItem.back { listener(Msg.Back) }
    )

}